powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/database/model/converter/MapToJsonConverterTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.database.model.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wultra.core.http.common.headers.UserAgent;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * * Test for {@link MapToJsonConverter}.
+ *
+ * @author Jan Dusil, jan.dusil@wultra.com
+ */
+
+class MapToJsonConverterTest {
+    private MapToJsonConverter converter;
+
+    @BeforeEach
+    void setUp() {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        converter = new MapToJsonConverter(objectMapper);
+    }
+
+    /**
+     * Tests the conversion of a map containing serializable objects, including a complex object,
+     * to a JSON string for database storage.
+     * Verifies that the complex object is correctly serialized and the resulting JSON string
+     * matches the expected format.
+     */
+    @Test
+    void testConvertToDatabaseColumnSerializableObject() {
+        Map<String, Serializable> testMap = new HashMap<>();
+        testMap.put("key1", "value1");
+        final String exampleRequestUserAgent = "PowerAuthNetworking/1.1.7 (en; cellular) com.wultra.app.MobileToken.wtest/2.0.0 (Apple; iOS/16.6.1; iphone12,3)";
+        final UserAgent.Device device = UserAgent.parse(exampleRequestUserAgent);
+        testMap.put("key2", device);
+
+        final String jsonResult = converter.convertToDatabaseColumn(testMap);
+        assertNotNull(jsonResult);
+        assertFalse(jsonResult.isEmpty());
+        assertEquals("{\"key1\":\"value1\"," +
```

Comment:
You may use multiline string if you want to.

Reply:
I don't think I can, then it adds newline chars

Reply:
🤔 Yes, you are right, that in that case you have to use `.replace("\n", "")`, which is sad. My idea was to avoid escaping. But it is just a minor issue to consider.